src/object/schema_system_catalog_install_query_spec.cpp

```diff
@@ -983,24 +983,25 @@ sm_define_view_authorization_spec (void)
 	      "WHERE "
 		"[u].[name] = CURRENT_USER"
 	    ") "
-	  "OR {[c].[class_of]} SUBSETEQ ("
+	  "OR {[a].[grantee].[name]} SUBSETEQ ("
 	      "SELECT "
-		"SUM (SET {[au].[object_of]}) "
+	        "SET {CURRENT_USER} + COALESCE (SUM (SET {[t].[g].[name]}), SET {}) "
 	      "FROM "
-		/* CT_CLASSAUTH_NAME */
-		"[%s] AS [au] "
+		/* AU_USER_CLASS_NAME */
+		"[%s] AS [u], TABLE ([u].[groups]) AS [t] ([g]) "
 	      "WHERE "
-		"{[au].[grantee].[name]} SUBSETEQ ("
-		    "SELECT "
-		      "SET {CURRENT_USER} + COALESCE (SUM (SET {[t].[g].[name]}), SET {}) "
-		    "FROM "
-		      /* AU_USER_CLASS_NAME */
-		      "[%s] AS [u], TABLE ([u].[groups]) AS [t] ([g]) "
-		    "WHERE "
-		      "[u].[name] = CURRENT_USER"
-		  ") "
-		"AND [au].[auth_type] = 'SELECT'"
-	    ") ) "
+		"[u].[name] = CURRENT_USER"
+	      ") "
+	  "OR {[a].[grantor].[name]} SUBSETEQ ("
+	      "SELECT "
+	        "SET {CURRENT_USER} + COALESCE (SUM (SET {[t].[g].[name]}), SET {}) "
+	      "FROM "
+		/* AU_USER_CLASS_NAME */
+		"[%s] AS [u], TABLE ([u].[groups]) AS [t] ([g]) "
+	      "WHERE "
+		"[u].[name] = CURRENT_USER"
+	      ") "
```

Comment:
```suggestion
	  "OR {[a].[grantee].[name]} SUBSETEQ [membership].[allowed_users] "
	  "OR {[a].[grantor].[name]} SUBSETEQ [membership].[allowed_users] "
          ...
```
여러 곳에서 해당 sub-query 부분이 중복되는데, sub-query 한 번만 사용하는 방법은 어떨까요? 유지보수에도 도움이 될 거 같습니다.
- 예시
  ```SELECT SET {CURRENT_USER} + COALESCE (SUM (SET {[t].[g].[name]}), SET {}) AS allowed_users FROM db_user AS [u], TABLE ([u].[groups]) AS [t] ([g]) WHERE [u].[name] = CURRENT_USER) AS [membership]```
  이후 `[membership].allowed_users` 의 subset 인지 확인

Reply:
코드 리뷰 감사합니다.
아래 현익님의 comment를 참고하여 중복된 SUBSETEQ 사용을 교집합 연산자(*)와 SETNEQ 기반으로 수정 했습니다.

Reply:
다른 함수에서도 같은 로직이 있는데, 해당 함수만 바꾸면 혼동이 있지 않을까요?
같이 리팩토링 하는 작업이 필요해 보입니다.

Reply:
#### db_auth 뷰 카탈로그만 grantor–grantee 관계를 보여주고, 다른 뷰는 기존대로 유지하도록 논의 됐습니다.

#### db_auth 외 뷰 카탈로그는 기존 동작을 유지합니다.
1. DBA 및 DBA 멤버
2. Owner 및 Owner 멤버
3. CURRENT_USER(+그룹) 이 객체([c].[class_of])에 대해 SELECT 권한을 부여받은 경우

#### 참고
- db_auth의 조건식을 다른 뷰에 적용할 경우, (1) DBA와 (2) Owner까지만 교집합 방식으로 묶을 수 있습니다.
- 필요 시 이 부분은 별도 PR로 진행하는 것이 좋아보입니다.

```sql
...
AND (
  /* (1) DBA, (2) OWNER */
  {'DBA', [c].[owner].[name]} * (
    SELECT
      SET {CURRENT_USER} + COALESCE(SUM(SET {[t].[g].[name]}), SET {})
    FROM
      /* AU_USER_CLASS_NAME */
      [%s] AS [u],
      TABLE([u].[groups]) AS [t]([g])
    WHERE
      [u].[name] = CURRENT_USER
  ) SETNEQ {}

  /* (3) CURRENT_USER가 해당 객체에 SELECT 권한이 있는 경우 */
  OR {[c].[class_of]} SUBSETEQ (
    SELECT
      SUM(SET {[au].[object_of]})
    FROM
      /* CT_CLASSAUTH_NAME */
      [%s] AS [au]
    WHERE
      {[au].[grantee].[name]} SUBSETEQ (
        SELECT
          SET {CURRENT_USER} + COALESCE(SUM(SET {[t].[g].[name]}), SET {})
        FROM
          /* AU_USER_CLASS_NAME */
          [%s] AS [u],
          TABLE([u].[groups]) AS [t]([g])
        WHERE
          [u].[name] = CURRENT_USER
      )
      AND [au].[auth_type] = 'SELECT'
    )
  )
...
```